src/executor/graph_executor.cc

```diff
@@ -321,16 +321,24 @@ void GraphExecutor::Init(nnvm::Symbol symbol,
                          const std::vector<OpReqType>& grad_req_type,
                          const std::vector<NDArray>& aux_states,
                          Executor* shared_exec) {
+  std::vector<SharedStorageEntry> shared_pool;
+  if (shared_exec != nullptr) {
+     for (auto& nd : dynamic_cast<GraphExecutor*>(shared_exec)->data_pool_) {
+       size_t bytes = nd.shape().Size() * mshadow::mshadow_sizeof(nd.dtype());
+       shared_pool.emplace_back(nd.ctx().dev_id, bytes); 
```

Comment:
There used to be a struct like this

https://github.com/dmlc/mxnet/blob/v0.8.0/src/symbol/graph_executor.h#L60
https://github.com/dmlc/mxnet/blob/v0.8.0/src/symbol/graph_executor.h#L60